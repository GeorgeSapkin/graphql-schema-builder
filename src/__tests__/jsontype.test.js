'use strict';

const graphql = require('graphql');

const {
  Kind: {
    BOOLEAN,
    FLOAT,
    INT,
    LIST,
    OBJECT,
    STRING
  }
} = graphql;

const {
  GraphQLJSON: _GraphQLJSON,

  parseLiteral: _parseLiteral
} = require('../jsontype');

const GraphQLJSON  = _GraphQLJSON(graphql);
const parseLiteral = _parseLiteral(graphql);

describe('GraphQLJSON', () => {
  const obj = { a: 1 };
  it('serialize works', () => {
    expect(GraphQLJSON.serialize(obj)).toMatchSnapshot();
  });

  it('parseValue works', () => {
    expect(GraphQLJSON.parseValue(obj)).toMatchSnapshot();
  });
});

describe('parseLiteral', () => {
  describe('returns', () => {
    it('boolean value', () => {
      expect(parseLiteral({
        kind:  BOOLEAN,
        value: true
      })).toMatchSnapshot();
    });

    it('string value', () => {
      expect(parseLiteral({
        kind:  BOOLEAN,
        value: 'abc'
      })).toMatchSnapshot();
    });

    it('float value when FLOAT', () => {
      expect(parseLiteral({
        kind:  FLOAT,
        value: '1.23'
      })).toMatchSnapshot();
    });

    it('float value when INT', () => {
      expect(parseLiteral({
        kind:  INT,
        value: '1'
      })).toMatchSnapshot();
    });

    it('list value', () => {
      expect(parseLiteral({
        kind:  LIST,
        values: [{
          kind:  FLOAT,
          value: '1.23'
        }, {
          kind:  STRING,
          value: 'abc'
        }]
      })).toMatchSnapshot();
    });

    it('object value', () => {
      expect(parseLiteral({
        kind:   OBJECT,
        fields: [{
          name: {
            value: 'a'
          },
          value: {
            kind:  STRING,
            value: 'abc'
          }
        }, {
          name: {
            value: 'b'
          },
          value: {
            kind:   OBJECT,
            fields: [{
              name: {
                value: 'c'
              },
              value: {
                kind:  BOOLEAN,
                value: false
              }
            }]
          }
        }]
      })).toMatchSnapshot();
    });

    it('null', () => {
      expect(parseLiteral({
        kind:  123,
        value: 456
      })).toMatchSnapshot();
    });
  });
});
