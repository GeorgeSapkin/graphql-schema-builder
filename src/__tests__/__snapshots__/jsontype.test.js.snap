// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphQLJSON parseValue works 1`] = `
Object {
  "a": 1,
}
`;

exports[`GraphQLJSON serialize works 1`] = `
Object {
  "a": 1,
}
`;

exports[`parseLiteral returns boolean value 1`] = `true`;

exports[`parseLiteral returns float value when FLOAT 1`] = `1.23`;

exports[`parseLiteral returns float value when INT 1`] = `1`;

exports[`parseLiteral returns list value 1`] = `
Array [
  1.23,
  "abc",
]
`;

exports[`parseLiteral returns null 1`] = `null`;

exports[`parseLiteral returns object value 1`] = `
Object {
  "a": "abc",
  "b": Object {
    "c": false,
  },
}
`;

exports[`parseLiteral returns string value 1`] = `"abc"`;
